src/consumer/base_consumer.rs

```diff
@@ -39,7 +39,11 @@ pub(crate) unsafe extern "C" fn native_commit_cb<C: ConsumerContext>(
     } else {
         Ok(())
     };
-    let tpl = TopicPartitionList::from_ptr(offsets);
+    let tpl = if offsets.is_null() {
+        TopicPartitionList::new()
```

Comment:
Thanks for tracking this down. This looks like a memory leak though thanks to the `mem::forget` call on L49?

Reply:
Good point. I've added some logic to handle both cases.